src/main/java/org/openpnp/machine/reference/feeder/ReferenceDragFeeder.java

```diff
@@ -149,45 +176,69 @@ public void feed(Nozzle nozzle) throws Exception {
         // feedEndLocation and pickLocation. pickLocation will be saved
         // for the pick operation while feed start and end are used
         // here and then discarded.
-        Location feedStartLocation = this.feedStartLocation;
-        Location feedEndLocation = this.feedEndLocation;
         pickLocation = this.location;
-        if (visionOffset != null) {
-            feedStartLocation = feedStartLocation.subtract(visionOffset);
-            feedEndLocation = feedEndLocation.subtract(visionOffset);
-            pickLocation = pickLocation.subtract(visionOffset);
-        }
 
-        // Move the actuator to the feed start location.
-        actuator.moveTo(feedStartLocation.derive(null, null, Double.NaN, Double.NaN));
+        if (feededCount == 0) {
```

Comment:
This is a big change, and I don't really understand what it does. Can you explain the changes to me please?

Reply:
I like it!

Reply:
I didnt made main correction to backoff movement. It should be done without backslash, but i didnt find any move function without backslash. It will be nice to do it. Now iam trying to make mgrl autoFeeders, maybe later i will return to drag feeder improvements.

About is0402, it depends on package name, not part name. If user want use drag feeder with 0402, he can rename package). 0201 easy to add same way.
Instead of adding this checkbox, we can add "parts per drag" input and it add more features. For example drag 4 sop8 ICs, then pick it one by one, then drag again.

Reply:
As Jason said, doing this based on a package name is a bad idea. We for example use metrical packages, e.g. 1005 for your 0402. A "parts per drag" seems to be a good option, defined in each feeder instance. 

@aneox For our "Heapfeeder", https://www.youtube.com/watch?v=rYFK_RLNU-g we also had to move the head without backlash compensation. 

This is what we use in our code, it's a hack, but it works: 
```
    // skips the backlash compensation
    // TODO 4: we assume millimeters here
    // TODO 4: only GcodeDriver is supported
    private void directMoveTo(HeadMountable generalHm, Location location) throws Exception{
    	directMoveTo(generalHm, location, getMachine().getSpeed());
    }
    
    private void directMoveToAtSafeZ(HeadMountable generalHm, Location location) throws Exception{
    	generalHm.moveToSafeZ();
    	double safeZ = generalHm.getLocation().getZ();
    	directMoveTo(generalHm, location.derive(null,  null,  safeZ, null));
    	directMoveTo(generalHm, location);
    }
    
    private void directMoveTo(HeadMountable generalHm, Location location, double speed) throws Exception{
    	ReferenceHeadMountable hm = (ReferenceHeadMountable) generalHm;
    	
    	ReferenceDriver driver =  getMachine().getDriver();
    	Location lHead = location.subtract(hm.getHeadOffsets()).convertToUnits(LengthUnit.Millimeters);

    	if(driver instanceof GcodeDriver) {
    		GcodeDriver d = (GcodeDriver) driver;

    		// TODO 4: fix this hardcoded axis handling
    		if(hm instanceof ReferenceCamera) {
        		d.moveTo(hm, lHead.derive(null,  null,  0.0,  null), speed, true); // TODO 3: fix save-z constant
    		}else {
    			d.moveTo(hm, lHead, speed, true);
    		}
    		
    	} else {
    		hm.moveTo(location);
    	}
    }
    
    // wrapper function, using backlash compensation, if configured
    private void preciseMoveTo(ReferenceHeadMountable hm, Location location) throws Exception{
    	hm.moveTo(location);
    }
```

We extended the interface in the GCodeDriver by a `skipBacklashCompensation`: 
```
	public void moveTo(ReferenceHeadMountable hm, Location location, double speed, boolean skipBacklashCompensation)
			throws Exception {
```

You can find the current code there:
https://github.com/KarlZeilhofer/openpnp/tree/feature/heapfeeder

Reply:
@KarlZeilhofer thank you moveTo with skipBacklashCompensation good idea!